tests/ci/docker_images/linux-x86/ubuntu-22.04_gcc-12x/Dockerfile

```diff
@@ -16,6 +16,9 @@ RUN set -ex && \
     rm -rf /var/lib/apt/lists/* && \
     rm -rf /tmp/*
 
+# Create alias to use cpp-12 preprocessor as default. MySQL relies on cpp being available.
+RUN update-alternatives --install /usr/bin/cpp cpp-bin /usr/bin/cpp-12 100
```

Comment:
Although build scripts should always use the `CC` and `CXX` environment variables, we should use `update-alternatives` to also setup `gcc-12` and `g++-12` for consistency with `cpp`. This is especially true since we're using this image to build third-party projects.

Reply:
Updated 👍